configstack/stack_plan.go

```diff
@@ -0,0 +1,167 @@
+	// stacks if one dependency has changes
+	detailedExitCode := util.ListContainsElement(terragruntOptions.TerraformCliArgs, "-detailed-exitcode")
+	if detailedExitCode {
+		util.RemoveElementFromList(terragruntOptions.TerraformCliArgs, "-detailed-exitcode")
+	}
+
+	results := make([]moduleResult, 0, len(stack.Modules))
+	err := RunModulesWithHandler(stack.Modules, getResultHandler(detailedExitCode, &results), NormalOrder)
+	printSummary(terragruntOptions, results)
+
+	// If there is no error, but -detail-exitcode is specified, we return an error with the number of changes.
+	if err == nil && detailedExitCode {
+		sum := 0
+		for _, status := range results {
+			sum += status.NbChanges
+		}
+		err = countError{sum}
+	}
+
+	return err
+}
+
+// Returns the handler that will be executed after each completion of `terraform plan`
+func getResultHandler(detailedExitCode bool, results *[]moduleResult) ModuleHandler {
+	return func(module TerraformModule, output string, err error) error {
+		warnAboutMissingDependencies(module, output)
+		if exitCode, convErr := shell.GetExitCode(err); convErr == nil && detailedExitCode && exitCode == CHANGE_EXIT_CODE {
+			// We do not want to consider CHANGE_EXIT_CODE as an error and not execute the dependants because there is an "error" in the dependencies.
+			// CHANGE_EXIT_CODE is not an error in this case, it is simply a status. We will reintroduce the exit code at the very end to mimic the behaviour
+			// of the native terrafrom plan -detailed-exitcode to exit with CHANGE_EXIT_CODE if there are changes in any of the module in the stack.
+			err = nil
+		}
+
+		message, count := extractSummaryResultFromPlan(output)
+
+		// We add the result to the result list (there is no concurrency problem because it is handled by the running_module)
+		*results = append(*results, moduleResult{module, err, message, count})
+
+		return err
+	}
+}
+
+// Print a little summary of the plan execution
+func printSummary(terragruntOptions *options.TerragruntOptions, results []moduleResult) {
+	fmt.Fprintf(terragruntOptions.Writer, "%s\nSummary:\n", separator)
+	for _, result := range results {
+		errMsg := ""
+		if result.Err != nil {
+			errMsg = fmt.Sprintf(", Error: %v", result.Err)
+		}
+
+		fmt.Fprintf(terragruntOptions.Writer, "    %v : %v%v\n", result.Module.Path, result.Message, errMsg)
+	}
+}
+
+// Check the output message
+func warnAboutMissingDependencies(module TerraformModule, output string) {
+	if strings.Contains(output, ": Resource 'data.terraform_remote_state.") {
+		var dependenciesMsg string
+		if len(module.Dependencies) > 0 {
+			dependenciesMsg = fmt.Sprintf(" contains dependencies to %v and", module.Config.Dependencies.Paths)
+		}
+		module.TerragruntOptions.Logger.Printf("%v%v refers to remote state "+
+			"you may have to apply your changes in the dependencies prior running terragrunt plan-all.\n",
+			module.Path,
+			dependenciesMsg,
+		)
+	}
+}
```

Comment:
This method could return a string or list of strings with warnings in them rather than logging directly. That way, you could write a unit test for it.